Source/JavaScriptCore/runtime/JSCJSValueInlines.h

```diff
@@ -112,15 +112,19 @@ inline size_t JSValue::toTypedArrayIndex(JSGlobalObject* globalObject, const cha
     RELEASE_AND_RETURN(scope, outputOffset + static_cast<size_t>(static_cast<uint32_t>(JSC::toInt32(d - inputOffset))));
 }
 
-// https://tc39.es/ecma262/#sec-tointegerorinfinity
-inline double JSValue::toIntegerOrInfinity(JSGlobalObject* globalObject) const
+// https://tc39.es/proposal-temporal/#sec-temporal-tointegerwithoutrounding
+inline double JSValue::toIntegerWithoutRounding(JSGlobalObject* globalObject) const
 {
     if (isInt32())
         return asInt32();
     double d = toNumber(globalObject);
-    if (std::isnan(d) || !d)
-        return 0.0;
-    return trunc(d);
+    return (std::isnan(d) || !d) ? 0.0 : d;
```

Comment:
Do you need the isnan here? I thought double is falsy if 0, -0, or nan?

Reply:
It does seem necessary; if I remove it and also add `dataLogLnIf(std::isnan(result[i]), !result[i]);` before the `isInteger` check in the constructor, I see:
```
>>> new Temporal.Duration(NaN)
false
Exception: RangeError: Temporal.Duration properties must be integers
```

The C++ spec confirms this:
> A zero value, null pointer value, or null member pointer value is converted to `false`; any other value is converted to `true`.